glide-core/redis-rs/redis/src/cluster_routing.rs

```diff
@@ -401,6 +418,13 @@ enum RouteBy {
     Undefined,
 }
 
+/// Defines values' positions in a command line for commands which could be forwarded to multiple shards
+enum MultiShardValues {
+    NoValues,          // for example, `MGET key1 key2`
+    ValueAfterEachKey, // for example, `MSET key1 value1 key2 value2`
+    ValueAfterAllKeys, // for example, `JSON.MGET key1 key2 key3 value`
```

Comment:
The only reason that combine_and_sort_array_results still worked for JSON.MGET is because the path argument was the last one and then the length of keys_indices  was larger from the values vector and the zip ignored the last item. Again, very not readable and can be a point for bugs later on.
Please refactor it and add better description so it would be more clear and readable

Reply:
`combine_and_sort_array_results` should accept only vectors with indices that match the order of results in the final combined array. 

With the current `MGET` implementation, we can directly pass the indices vector from `MultipleNodeRoutingInfo::MultiSlot` because each key index corresponds to the index of the result in the final array. 

However, if we were to use `combine_and_sort_array_results` for `MSET`, for example, we would need to adjust the indices array accordingly. The same applies to `JSON.MGET`; you should modify this array or store a separate array specifically for result indices to maintain readability and correctness.